packages/web-components/src/menu/menu.ts

```diff
@@ -0,0 +1,204 @@
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: open-on-hover
+   */
+  @attr({ attribute: 'open-on-hover', mode: 'boolean' })
+  public openOnHover: boolean = false;
+
+  private _menu?: MenuList;
+  private _trigger?: HTMLElement;
+
+  public menuListContainer?: HTMLElement;
+
+  private cleanup?: () => void;
+
+  connectedCallback() {
+    super.connectedCallback();
+    document.addEventListener('click', this.handleDocumentClick);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    document.removeEventListener('click', this.handleDocumentClick);
+  }
+
+  public setPositioning(): void {
+    if (this.$fastController.isConnected && this._trigger && this.menuListContainer) {
+      this.cleanup = autoUpdate(this, this.menuListContainer, async () => {
+        const { middlewareData, x, y } = await computePosition(this._trigger!, this.menuListContainer!, {
+          placement: 'bottom',
+          strategy: 'fixed',
+          middleware: [
+            flip(),
+            size({
+              apply: ({ availableHeight, rects }) => {
+                this.menuListContainer?.style &&
+                  Object.assign(this.menuListContainer.style, {
+                    maxHeight: `${availableHeight}px`,
+                    width: `${rects.reference.width}px`,
+                  });
+              },
+            }),
+            hide(),
+          ],
+        });
+        if (middlewareData.hide?.referenceHidden) {
+          this.expanded = false;
+          return;
+        }
+
+        this.menuListContainer?.style &&
+          Object.assign(this.menuListContainer.style, {
+            position: 'fixed',
+            top: '0',
+            left: '0',
+            transform: `translate(${x}px, ${y}px)`,
+          });
+      });
+    }
+  }
+
+  // Declare menu as an observable property of type HTMLElement[]
+  @observable
+  public menu?: HTMLElement[];
+
+  // Method to update the menu when it changes
+  protected menuChanged(): void {
+    if (this.$fastController.isConnected && this.menu && this.menu.length) {
+      this._menu = this.menu[0] as MenuList;
+    }
+  }
+
+  // Declare trigger as an observable property of type HTMLElement[]
+  @observable
+  public trigger?: HTMLElement[];
+
+  // Method to update the trigger when it changes
+  protected triggerChanged(prev: HTMLElement[] | void, next: HTMLElement[]): void {
+    if (prev !== next) {
+      if (Array.isArray(prev) && prev.length) {
+        this.cleanupTriggerEventListeners(prev[0]);
+      }
+
+      if (next && next.length) {
+        this._trigger = next[0];
+
+        this._trigger.setAttribute('aria-haspopup', 'true');
+        this._trigger.setAttribute('aria-expanded', `${this.expanded}`);
+
+        // Add new event listeners
+        if (this.openOnHover) {
+          this._trigger.addEventListener('mouseover', this.toggleExpanded);
+        } else {
+          this._trigger.addEventListener('click', this.toggleExpanded);
+        }
```

Comment:
Can these be added directly to the template given certain conditions? The document listener cannot but each of these could be. `@mouseover="${x => x...}"`

Reply:
These event listeners are being applied to the MenuButtons (this._trigger). So we would be adding these to the MenuButton template and then emitting an event that the Menu would need to listen for? But at that point we would still need to attach event listeners no? I would have thought that was the right approach if MenuButton wasn't so tightly coupled with the Menu as in if we expected other components would need to listen for these events?

Reply:
@brianchristopherbrady you're right - I think I started here and then looked at the template. Since the trigger is slotted this is 100% accurate.

Reply:
And would also mean we might want to move attr openOnHover to the MenuButton? Otherwise we would be firing off mouseover events unnecessarily? so we can say..

```ts
  if(this.openOnHover) { emit mouseoverEvent }
```